bin/test

```diff
@@ -0,0 +1,8 @@
+#! /bin/bash
+
+printf "\nrunning specs with 'crystal spec'\n\n"
+crystal spec
+printf "\nbuilding shards with 'shards build'\n\n"
+shards build
+printf "\nformatting code with 'crystal tool format src spec'\n\n"
+crystal tool format src spec
```

Comment:
I think this is a good change! Just two thoughts:

- I think crystal tool format should be run on all the code instead of just the specs since the CI specs will fail if there are any changes.
- And do you think we should have `crystal spec` run at the end in case `crystal tool format` makes changes that break the specs?

Reply:
I'll tweak the script to it runs ```crystal tool format``` on the entire project, and I can see the value in running ```crystal spec``` as the final command. I'm thinking of a scenario when that would happen though. 

It would probably be a case where a new contributor makes a change and writes some new specs with formatting issues, runs ```./bin/test``` and then gets failures on the final ```crystal spec``` command. This sounds like a valid scenario and I think we would want the specs to fail in that case.

Reply:
Actually I'm thinking that we wouldn't want to run the formatter on the lib directory. Thoughts?

Reply:
You're right we don't want to run the formatter on the lib directory, but the `crystal tool format` command excludes the lib directory by default. You can see [the code here](https://github.com/crystal-lang/crystal/blob/e11ee26c7cfc5c1e1982e84d318af4d063a8de72/src/compiler/crystal/command/format.cr#L18)